src/constants/screens.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Mapping of a breakpoint name to the lower-bound of its screen-width range
+ * @type {Map<string, number>}
+ */
+export const SCREEN_SIZES = new Map()
+SCREEN_SIZES.set('2xl', 1536)
+SCREEN_SIZES.set('xl', 1280)
+SCREEN_SIZES.set('lg', 1024)
+SCREEN_SIZES.set('md', 768)
+SCREEN_SIZES.set('sm', 640)
```

Comment:
I definitely understand the desire to use a map here, but an object would give us better type support and allow us to type the argument of the `isScreen` method, for example. Something like this:

```js
/**
 * @readonly
 * @enum {number} The pixel width of the screen
 */
export const screens = {
  sm: 640,
  md: 768,
  // ... and so on
}
```

Reply:
I used `Map` because it guarantees read-order of keys will be the same as insert order. This guarantee was useful because `getBreakpointName` relies on the screen widths being ordered.

Map offers the same typing interface as object with two generics `K, V` representing the key and value. JetBrains is able to infer arguments to both `.get` and `.set` methods of this map using the `@type` annotation.

Reply:
Where is getBreakpointName used?

Reply:
It is used further down in this file.